grc/gui/Utils.py

```diff
@@ -70,28 +71,35 @@ def align(value):
         return align(coor)
 
 
-def num_to_str(num):
+def num_to_str(num) -> str:
     """ Display logic for numbers """
-    def eng_notation(value, fmt='g'):
-        """Convert a number to a string in engineering notation.  E.g., 5e-9 -> 5n"""
-        template = '{:' + fmt + '}{}'
-        magnitude = abs(value)
-        for exp, symbol in zip(range(9, -15 - 1, -3), 'GMk munpf'):
-            factor = 10 ** exp
-            if magnitude >= factor:
-                return template.format(value / factor, symbol.strip())
-        return template.format(value, '')
-
-    if isinstance(num, numbers.Complex):
-        num = complex(num)  # Cast to python complex
-        if num == 0:
-            return '0'
-        output = eng_notation(num.real) if num.real else ''
-        output += eng_notation(num.imag, '+g' if output else 'g') + \
-            'j' if num.imag else ''
-        return output
-    else:
-        return str(num)
+
+    def clean_e(unclean: str) -> str:
+        """
+        Removes the + from 1.2e+09, and the 0 from both 1.2e-09 and 1.2e09
+        """
+        return unclean\
+            .replace("e-0", "e-")\
+            .replace("e+0", "e+")\
+            .replace("e+", "e")\
+            .strip()
+
+    if numpy.iscomplex(num):
+        intreal = int(numpy.real(num))
+        intimag = int(numpy.imag(num))
+        if intreal == numpy.real(num) and \
+           intimag == numpy.imag(num) and \
+           abs(num) < 2**0.5 * 10**6:
```

Comment:
Could it just be based on max(real, imaj)? Using abs could end up causing some strange break points. Of course, this only happens when there's a pretty weird complex number in the first place.

Reply:
Yeah, I thought about that, but came down to the point where I said "If the magnitude > 1.4e6, you'll always want _-notation for both parts, both if they are very different in magnitude, or when they are roughly the same". But since this is for integer im, re only, anyways, not so sure it makes too much sense.

Happy to change it if you have strong feelings about it, otherwise would leave it like it is.